packages/block/src/block.ts

```diff
@@ -340,10 +354,13 @@ export class Block {
       }
     }
 
-    const transactionsTrie = await Block.genTransactionsTrieRoot(txs)
+    const transactionsTrie = await Block.genTransactionsTrieRoot(
+      txs,
+      new Trie({ common: opts?.common })
+    )
```

Comment:
Ah, ok, seeing a bit where this was coming from with the `trie` option made mandatory, should nevertheless not be necessary. Here we pass over the new trie and then that's it.